components/tikv_util/src/interval_driver.rs

```diff
@@ -4,61 +4,60 @@ use std::io;
 use std::result::Result;
 use std::sync::mpsc::{self, Sender};
 use std::thread::{Builder as ThreadBuilder, JoinHandle};
-use std::time::{Duration, Instant};
-
-use file_system::flush_io_metrics;
-
-use crate::engine::KvEngine;
-use crate::engines::Engines;
-use crate::raft_engine::RaftEngine;
+use std::time::Duration;
 
 const DEFAULT_FLUSH_INTERVAL: Duration = Duration::from_millis(10_000);
-const FLUSHER_RESET_INTERVAL: Duration = Duration::from_millis(60_000);
 
-pub struct MetricsFlusher<K: KvEngine, R: RaftEngine> {
-    pub engines: Engines<K, R>,
+pub struct IntervalDriver {
```

Comment:
I think `MetricsFlusher` is better. It's only used for flushing metrics. Other interval tasks are scheduled in their own worker.

Reply:
I don't really like it, because the class itself has no involvement in actual logic. If you think the name isn't friendly enough, it is actually only used in server.rs, all the task definitions are self-explanatory.

Reply:
I mean move it into `server` instead of `tikv_util` and make `MetricsFlusher` which implements RunnableWithTimer or just keep the origin implementation.  Now the `IntervalDriver` is quite duplicated with `tikv_util::Worker`.

Reply:
We can't do that because metrics task are defined in submodules. I'll try the reuse to see if they look better.

Reply:
You don't need metrics task in that way

Reply:
You mean fusing all those module logic into server.rs? Be aware that each metrics flushing routine has their own states, exposing them to server.rs would be chaotic.

Reply:
On a second thought, I decided to make some less generic wrapper for those states, so we can hardcode them into a runner in server.rs.

Reply:
Yes, I mean the level of `server` mod, namely, a runner in `server.rs`